app/models/setting.rb

```diff
@@ -45,7 +45,7 @@ def validate(record)
   validates :value, :array_hostnames_ips => true, :if => proc { |s| ARRAY_HOSTNAMES.include? s.name }
   validates :value, :email => true, :if => proc { |s| EMAIL_ATTRS.include? s.name }
   before_save :clear_value_when_default
-  before_save :encrypt_url_if_password_present, :if => proc { |s| s.value.is_a?(String) && s.value.start_with?('http') }
+  before_save :encrypt_url_if_password_present, :if => proc { |s| s.name == "http_proxy" && s.value.is_a?(String) && s.value.start_with?('http') }
```

Comment:
The additional conditions `&& s.value.is_a?(String) && s.value.start_with?('http')` appear redundant since we already have `HttpURLValidator` validation on the `http_proxy` setting (see `f_foreman_settings_general.rb:43`). 

The validator ensures that any present value is already a valid HTTP/HTTPS URL string, so this could be simplified to:

```ruby
before_save :encrypt_url_if_password_present, :if => proc { |s| s.name == "http_proxy" && s.value.present? }
```

The `present?` check is still needed to avoid calling the encryption method on nil/blank values.